src/ucp/core/ucp_ep.inl

```diff
@@ -49,6 +60,10 @@ static inline ucp_rsc_index_t ucp_ep_num_lanes(ucp_ep_h ep)
 static inline ucp_rsc_index_t ucp_ep_md_index(ucp_ep_h ep, ucp_lane_index_t lane)
 {
     ucp_context_h context = ep->worker->context;
+#if ENABLE_ASSERT
```

Comment:
no nee to use "ENABLE_ASSERT" here, can avoid using the "rsc_index" variable:
`ucs_assert(ucp_ep_get_rsc_index(ep, lane) != UCP_NULL_RESOURCE)`